tags/interceptors.go

```diff
@@ -56,7 +57,9 @@ func (w *wrappedStream) Context() context.Context {
 func (w *wrappedStream) RecvMsg(m interface{}) error {
 	err := w.ServerStream.RecvMsg(m)
 	// We only do log fields extraction on the single-reqest of a server-side stream.
-	if !w.info.IsClientStream {
+	if !w.info.IsClientStream || w.opts.requestFieldsFromInitial && w.initial {
+		w.initial = false
```

Comment:
Presumably `RecvMsg` will never be called concurrently for the same stream twice at the same time. If this assumption is incorrect, this may need rework.

Reply:
No, it is called from the same goroutine as it is reading msgs from the stream. So it is safe.